### PR TITLE
Autoscaling example

### LINKED ISSUE #41: `kubernetes.prometheus.autoscale` example
Use `stream.active.received` metric on simple example, such as `tcp.echo`, to illustrate `kubernetes` horizontal pod autoscaling via prometheus.

### PR DESCRIPTION
**Pull request by attilakreiner**
fixes #41

### CODE REVIEW

File: kubernetes.prometheus.autoscale/README.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+```
+
+Open 21 more connections to zilla.
+
+```bash
+# Open 21 more instances of netcat in the background.
+$ for i in `seq 1 21`; do nc localhost 8080 &; done
+[77] 77777
+[78] 77776
+[79] 77775
```
Comment (jfallows) [inline kubernetes.prometheus.autoscale/README.md:162]: PIDs should increase I think.

File: kubernetes.prometheus.autoscale/README.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+Prometheus API:
+{
+...
+        "metric": {
+          "__name__": "stream_active_received",
+        },
+        "value": [
+          "0"
+...
+}
```
Comment (jfallows) [inline kubernetes.prometheus.autoscale/README.md:243]: When running the example locally, I'm getting an unusual value in the list after scale-in.
```
Prometheus API:
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "stream_active_received",
          "binding": "tcp_server0",
          "endpoint": "prometheus",
          "exported_namespace": "example",
          "instance": "10.1.0.180:9090",
          "job": "zilla",
          "namespace": "zilla-autoscaling",
          "pod": "zilla-6db8d879f5-66fgt",
          "service": "zilla"
        },
        "value": [
          1682709653.16,
          "0"
        ]
      }
    ]
  }
}
```

**Review by jfallows — CHANGES_REQUESTED** [review]
Do we need the `_total` suffix to follow convention for our prometheus exporter, since we are totaling across cores during export?

File: kubernetes.prometheus.autoscale/README.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+### Setup
+
+The `setup.sh` script:
+- installs Zilla, Prometheus and Prometheus Adapter to the Kubernetes cluster with helm and waits for the pods to start up
+- sets up horizontal pod autoscaling
+- starts port forwarding
+
+```bash
+$ ./setup.sh
++ helm install zilla-autoscaling chart --namespace zilla-autoscaling --create-namespace --wait
```
[the remark this answers is not in the document]
Comment (attilakreiner) [reply kubernetes.prometheus.autoscale/README.md]: done

File: kubernetes.prometheus.autoscale/README.md (hunk 1 above)
Comment (attilakreiner) [reply kubernetes.prometheus.autoscale/README.md]: fixed

File: kubernetes.prometheus.autoscale/README.md (hunk 2 above)
Comment (attilakreiner) [reply kubernetes.prometheus.autoscale/README.md:247]: That's just how prometheus returns values like `"value": [ <timestamp>, <value> ]`. I updated the README showing this with comments.

**Comment by attilakreiner** [comment]
> Do we need the `_total` suffix to follow convention for our prometheus exporter, since we are totaling across cores during export?

We apply the `_total` suffix to counters following the prometheus naming convention (in accordance with the open telemetry prometheus exporter):

https://prometheus.io/docs/practices/naming/
> ...an accumulating count has total as a suffix, in addition to the unit if applicable.
> http_requests_total (for a unit-less accumulating count)
> process_cpu_seconds_total (for an accumulating count with unit)

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/translator/prometheus
> Counter | Append _total | system.processes.created ==> system_processes_created_total

https://github.com/aklivity/zilla/blob/7d4976557af634e5e5fed2f232d56f2ff2d2053e/incubator/exporter-prometheus/src/main/java/io/aklivity/zilla/runtime/exporter/prometheus/internal/descriptor/PrometheusMetricDescriptor.java#L70
```
            if (metric.kind() == COUNTER)
            {
                result += "_total";
            }
```

The metric `stream.active.received` is not a counter but a gauge so all the transformation we apply to the name is replacing dots with underscores: `stream_active_received`.

**Review by jfallows — APPROVED** [review]

**Pull request merged**